File: sourcecode/testdata/sample.php
<?php

/** Say hello to the given name. */
function greet(string $name): string
{
    return "hello $name";
}

/** A tiny server. */
class Server
{
    /** Start the server. */
    public function start(): bool
    {
        return true;
    }
}

interface Handler
{
}

trait Loggable
{
}

File: sourcecode/testdata/sample.py
"""Sample module."""


def greet(name):
    """Say hello to the given name."""
    return f"hello {name}"


class Server:
    """A tiny server."""

    def __init__(self, addr):
        self._addr = addr

    @property
    def addr(self):
        """Return the bind address."""
        return self._addr


def farewell(name):
    return f"goodbye {name}"
